### PR TITLE
Transition to gobin

### PR DESCRIPTION
**Pull request by sdake**
gobin compiles go software. It appears (although not verified) - to create
a per-build cache of source dependencies. The nice thing about this is that
as different tools in our base container are built, their dependencies don't
conflict.

### CODE REVIEW

File: docker/build-tools/Dockerfile [hunk 1]
```diff
@@ -88,38 +88,48 @@ ADD https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERS
 RUN unzip /tmp/protoc-${PROTOC_VERSION}-linux-x86_64.zip
 RUN mv /tmp/bin/protoc ${OUTDIR}/usr/bin
 
+# Install gobin. We use gobin to build all of our tools such that they remain
+# independent of their dependency versions.
+RUN GO111MODULE=off go build -o /usr/local/go/gobin github.com/myitcv/gobin
```
Comment (clarketm) [inline docker/build-tools/Dockerfile:93]:
```suggestion
RUN GO111MODULE=off go get github.com/myitcv/gobin
```

Comment (sdake) [reply docker/build-tools/Dockerfile]: ya I already had this in my local build. Although now mason is not building, and it was building locally.

Comment (clarketm) [reply docker/build-tools/Dockerfile]: It isn't building for me locally.

**Comment by sdake** [comment]
Have tested this container in istio, api, and docs repo.

**Comment by sdake** [comment]
@clarketm can you PTAL.

**Pull request merged**